src/averageAges.js

```diff
@@ -1,61 +1,37 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredMen = people.filter(man => man.sex === 'm'
+    && (!century || Math.ceil(man.died / 100) === century)
+  );
+
+  return filteredMen.reduce((acc, manAge) => (
+    acc + manAge.died - manAge.born
+  ), 0) / filteredMen.length;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomen = people.filter(woman => woman.sex === 'f'
+    && (!withChildren || people.some(person => person.mother === woman.name))
+  );
+
+  return filteredWomen.reduce((acc, womanAge) => (
+    acc + womanAge.died - womanAge.born
+  ), 0) / filteredWomen.length;
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => people.find(mother =>
+    child.mother === mother.name) && (!onlyWithSon || child.sex === 'm')
```

Comment:
as for me, you could add a few variables for easier readability. But this decision looks cute :)